src/gui/guiButton.cpp

```diff
@@ -490,6 +506,38 @@ void GUIButton::setImage(EGUI_BUTTON_IMAGE_STATE state, video::ITexture* image,
 	ButtonImages[stateIdx].SourceRect = sourceRect;
 }
 
+void GUIButton::setImage(video::ITexture* image)

```

Comment:
```
// PATCH
<new code>
// END PATCH
```

This file is imported from Irrlicht, hence changes should be visible easily in case there's an update.
It would also be great if this code would follow the code style guidelines, say pointer to variable -> `type *var`

Reply:
I've added the comments to guiButton.h/guiButton.cpp. I don't think the other files come from Irrlicht (correct me if I'm wrong), so I've left them as-is.

As for the formatting, do you want me to change the style of the entire file? I can do it, but I won't squash that change since it would make this patch harder to read.

Lastly, I noticed that while the pointer style that you want is used in the code style guidelines on the wiki it's never specifically mentioned. Just to be sure, is it implicit or is there a more up-to-date source that I should be following?

Reply:
Please don't change the style of this file, it should be in irrlicht's code style

Reply:
Ok then, I'll leave it as-is. @SmallJoker, can you confirm that my new changes are correct?

Reply:
> Lastly, I noticed that while the pointer style that you want is used in the code style guidelines on the wiki it's never specifically mentioned. Just to be sure, is it implicit or is there a more up-to-date source that I should be following?

https://dev.minetest.net/Code_style_guidelines:
>The coding style is based on the [Linux kernel code style](https://www.kernel.org/doc/html/latest/process/coding-style.html).

https://www.kernel.org/doc/html/latest/process/coding-style.html#spaces:
>When declaring pointer data or a function that returns a pointer type, the preferred use of `*` is adjacent to the data name or function name and not adjacent to the type name.

Reply:
Whoops, missed that. Thanks!